libr/debug/p/native/linux/linux_debug.c

```diff
@@ -611,6 +615,46 @@ RDebugPid *fill_pid_info(const char *info, const char *path, int tid) {
 	return pid_info;
 }
 
+RList *linux_pid_list(int pid, RList *list) {
+	list->free = (RListFree)&r_debug_pid_free;
+	DIR *dh;
+	struct dirent *de;
+	char *ptr, st, buf[1024], info[1024];
+	int i, thid, uid;
+	RDebugPid *pid_info;
+	dh = opendir ("/proc");
+	if (!dh) {
+		r_sys_perror ("opendir /proc");
+		r_list_free (list);
+		return NULL;
+	}
+	while ((de = readdir (dh))) {
+		memset (buf, 0, sizeof (buf));
+		memset (info, 0, sizeof (buf));
+		uid = 0;
+		st = ' ';
+		// For each existing pid file
+		i = atoi (de->d_name);
```

Comment:
you can define and assign in the same line

Reply:
They were reassigned in each iteration to avoid errors. Anyways, no longer needed.